test/extended/networking/livemigration.go

```diff
@@ -1,14 +1,680 @@
+	oc := exutil.NewCLIWithPodSecurityLevel("network-segmentation-e2e", admissionapi.LevelBaseline)
+	f := oc.KubeFramework()
+
 	InOVNKubernetesContext(func() {
-		It("dummy test [Suite:openshift/network/virtualization]", func() {
-			Expect(1).To(Equal(1))
+		var (
+			cs         clientset.Interface
+			nadClient  nadclient.K8sCniCncfIoV1Interface
+			virtClient *vmClient
+		)
+
+		BeforeEach(func() {
+			cs = f.ClientSet
+
+			var err error
+			nadClient, err = nadclient.NewForConfig(f.ClientConfig())
+			Expect(err).NotTo(HaveOccurred())
+			virtClient = newVirtClient(oc, "/tmp/")
+		})
+
+		Context("with user defined networks and persistent ips configured", func() {
+			const (
+				nadName           = "blue"
+				bindingName       = "passt"
+				udnCrReadyTimeout = 5 * time.Second
+				vmName            = "myvm"
+			)
+			var (
+				cidrIPv4 = "203.203.0.0/16"
+				cidrIPv6 = "2014:100:200::0/60"
+			)
+
+			DescribeTableSubtree("created using",
+				func(createNetworkFn func(netConfig networkAttachmentConfigParams)) {
+
+					DescribeTable("[Suite:openshift/network/virtualization] should keep ip", func(netConfig networkAttachmentConfigParams, resourceCmd func() string, opCmd func(cli *vmClient, vmNamespace, vmName string)) {
+						var err error
+						netConfig.namespace = f.Namespace.Name
+						// correctCIDRFamily makes use of the ginkgo framework so it needs to be in the testcase
+						netConfig.cidr = correctCIDRFamily(oc, cidrIPv4, cidrIPv6)
+						workerNodes, err := getWorkerNodesOrdered(cs)
+						Expect(err).NotTo(HaveOccurred())
+						Expect(len(workerNodes)).To(BeNumerically(">=", 2))
+
+						isDualStack := getIPFamilyForCluster(f) == DualStack
+
+						createNetworkFn(netConfig)
+
+						tmpl, err := template.New(f.Namespace.Name).Parse(resourceCmd())
+						Expect(err).To(Not(HaveOccurred()))
+						params := struct {
+							VMName                    string
+							VMNamespace               string
+							FedoraContainterDiskImage string
+							NetBindingName            string
+							NetworkName               string
+						}{
+							VMName:                    vmName,
+							VMNamespace:               f.Namespace.Name,
+							FedoraContainterDiskImage: "quay.io/kubevirt/fedora-with-test-tooling-container-disk:20241024_891122a6fc",
+						}
+						if netConfig.role == "primary" {
+							params.NetBindingName = bindingName
+						} else {
+							params.NetworkName = nadName
+						}
+
+						var cmd bytes.Buffer
+						Expect(tmpl.Execute(&cmd, params)).To(Succeed())
+						Expect(virtClient.createVM(f.Namespace.Name, vmName, cmd.String())).To(Succeed())
+						virtClient.waitForVMReadiness(f.Namespace.Name, vmName)
+
+						By("Retrieving addresses before test operation")
+						expectedAddresses := []string{}
+						if netConfig.role == "primary" {
+							expectedAddresses = addressFromGuest(virtClient, vmName)
+						} else {
+							expectedAddresses = addressFromStatus(virtClient, vmName)
+						}
```

Comment:
please turn this into an helper. It's more readable, and is invoked again a little bit below, when you compute the `obtainedAddresses` .